torchlpc/csrc/scan_cpu.cpp

```diff
@@ -0,0 +1,82 @@
+#include <torch/script.h>
+#include <torch/torch.h>
+#include <algorithm>
+#include <vector>
+#include <utility>
+
+template <typename scalar_t>
+void scan_cpu(const at::Tensor &input, const at::Tensor &weights, const at::Tensor &initials, at::Tensor &output)
```

Comment:
I'm sure you did it deliberately, but I don't see `output` or `output_ptr` being reassigned anywhere, so why not make them `const` as well for consistency?

Reply:
Hmm, I think you're right. I copied the signature from the tutorials. I guess they use `const` to emphasise mutable/immutable tensors. I'll check if `const` still works here.

Reply:
Actually the `output_ptr` should be constant otherwise it's not writable.